### PR TITLE
merge thrift

### PR DESCRIPTION
**Pull request by Leishunyu**
## What is the purpose of the change

https://github.com/apache/incubator-dubbo/issues/3960

## Brief changelog

Cover the original thrift

## Verifying this change

XXXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Make sure there is a [GITHUB_issue](https://github.com/apache/incubator-dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GITHUB issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Format the pull request title like `[Dubbo-XXX] Fix UnknownException when host config not exist #XXX`. Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/incubator-dubbo/tree/master/dubbo-test).
- [ ] Run `mvn clean install -DskipTests=false` & `mvn clean test-compile failsafe:integration-test` to make sure unit-test and integration-test pass.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/incubator-dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Comment by Leishunyu** [comment]
@beiwei30

File: dubbo-dependencies-bom/pom.xml [hunk 1]
```diff
@@ -106,7 +106,7 @@
         <consul_process_version>2.0.0</consul_process_version>
         <xmemcached_version>1.3.6</xmemcached_version>
         <cxf_version>3.1.15</cxf_version>
-        <thrift_version>0.8.0</thrift_version>
+        <thrift_version>0.11.0</thrift_version>
```
Comment (beiwei30) [inline dubbo-dependencies-bom/pom.xml]: pls. use the latest version. this version has security issue.

**Comment by Leishunyu** [comment]
> I think it's better to keep the two modules[1] together for a while now.
> [1] dubbo-rpc-thrift and dubbo-rpc-native-thrift
> 
> So we should upgrade the thrift version to `11` in dubbo-rpc-thrift

i think so, i change it

File: dubbo-rpc/dubbo-rpc-native-thrift/src/main/resources/META-INF/dubbo/internal/org.apache.dubbo.rpc.Protocol [hunk 2]
```diff
@@ -0,0 +1 @@
+nthrift=org.apache.dubbo.rpc.protocol.nativethrift.ThriftProtocol
```
Comment (beiwei30) [inline dubbo-rpc/dubbo-rpc-native-thrift/src/main/resources/META-INF/dubbo/internal/org.apache.dubbo.rpc.Protocol]: should **native-thrift** better?

File: dubbo-test/pom.xml [hunk 3]
```diff
@@ -126,6 +126,10 @@
             <groupId>org.apache.dubbo</groupId>
             <artifactId>dubbo-rpc-thrift</artifactId>
         </dependency>
+        <dependency>
```
Comment (beiwei30) [inline dubbo-test/pom.xml]: why do we need this?

File: dubbo-compatible/pom.xml [hunk 4]
```diff
@@ -62,11 +62,26 @@
...
+            <groupId>org.apache.dubbo</groupId>
+            <artifactId>dubbo-remoting-netty4</artifactId>
+            <version>${project.parent.version}</version>
+        </dependency>
         <dependency>
             <groupId>org.apache.dubbo</groupId>
             <artifactId>dubbo-rpc-thrift</artifactId>
             <version>${project.parent.version}</version>
         </dependency>
+        <dependency>
```
Comment (beiwei30) [inline dubbo-compatible/pom.xml]: why do we need this?

File: dubbo-rpc/dubbo-rpc-native-thrift/src/main/resources/META-INF/dubbo/internal/org.apache.dubbo.rpc.Protocol (hunk 2 above)
Comment (Leishunyu) [reply dubbo-rpc/dubbo-rpc-native-thrift/src/main/resources/META-INF/dubbo/internal/org.apache.dubbo.rpc.Protocol]: 👍  i do it

File: dubbo-test/pom.xml (hunk 3 above)
Comment (Leishunyu) [reply dubbo-test/pom.xml]: no,i remove it

File: dubbo-compatible/pom.xml (hunk 4 above)
Comment (Leishunyu) [reply dubbo-compatible/pom.xml]: no,i remove it

**Pull request merged**